File: 217 Contains Duplicate/contains_duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    nums.sort();
    for (i = 1; i < nums.length; i++) {
        if (nums[i - 1] == nums[i]) return true;
    }
    return false;
}; // Author: XC

File: 217 Contains Duplicate/contains_duplicate.cpp
/*	
 *	LeetCode Submissions by Xinyu Chen
 *	Contains Duplicate
 *	https://leetcode.com/problems/contains-duplicate/
 *	Runtime: 104 ms
 */
 
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        if (nums.size() < 2) return false;
        set<int> s;
        for (int n : nums) {
            if (s.find(n) != s.end()) return true;
            s.insert(n);
        }
        return false;
    }
};